enhancements/security/runtime-class-scc.md

```diff
@@ -0,0 +1,326 @@
+
+## Summary
+
+Enhance the Openshift SCC with a field called `RequiredRuntimeClasses` which gates users from specifying runtime classes in their Pod requests.
+
+## Motivation
+
+Enhancing SCC to be runtime class aware allows admins to customize how their users are able to access different runtime customizations.
+This is key in enabling new features like unprivileged builds, kata containers,
+and performance enhancements for latency sensitive workloads,
+while also providing protection from less privileged users gaining access to features they should not.
+
+### Goals
+
+- SCC should support specifying the runtime classes that must be specified in the pod request from a user.
+
+### Non-Goals
+
+- Changing the current SCC profiles to allow any extra runtime classes.
+- Changing behavior of the default runtime class.
+- Adding additional SCCs for various customzied behavior using RequiredRuntimeClasses.
+- Deciding how new features will use RequiredRuntimeClasses.
+
+## Proposal
+
+
+[RuntimeClasses][runtime-classes] is a feature in Kubernetes that allows a user to request a certain runtime configuration.
+CRI-O has enhanced this feature to not only branch on different runtimes, but customize the way the pod is run.
+With the customization of the way a pod is run comes risk of users allocating resources they should not be allowed to.
+This leads to the need for a way to gate different users from using different runtime classes,
+and SCC is a perfect component to do so.
+
+The proposal is to add a new field to the SCC: `RequiredRuntimeClasses`:
+
+The API for `RequiredRuntimeClasses` will follow the implementation of [RequiredDropCapabilities][required-drop-capabilities],
+where a specific SCC profile will be able to specify a set of runtime classes,
+and pods in that SCC must request a runtime class that matches that set.
+If a request for a runtime class comes in that doesn't match the required runtime classes, the request will be denied.
+If a request comes in that doesn't specify a runtime class, but the default runtime class (`""`) is not specified in `RequiredRuntimeClasses`
+the request will also be denied.
+
+However, instead of a customized type,
+or using a type from the Kubernetes API like [the Runtime Class structure][runtime-class-structure],
+`RequiredRuntimeClasses` will be a slice of strings.
+One can think of it as a list of [Handlers][handlers] of the runtime classes that are allowed.
+
+```
+type SecurityContextConstraints struct {
+    ...
+	// RequiredRuntimeClasses is a list of names of runtime classes (also called runtime handlers).
+	// The runtime class specified in a pod request must match one of the items in this list.
+	// To allow any runtime class, `*` may be specified.
+	// To ensure the default runtime class may be used, an empty string (`""`) can be specified.
+	// +listType=set
+	RequiredRuntimeClasses []string `json:"requiredRuntimeClasses" protobuf:"bytes,26,rep,name=requiredRuntimeClasses"`
```

Comment:
as empty list does not make sense anymore (`[""]` is now the default), you have to add `minLength: 1` as kubebuilder marker.

Reply:
I believe I did this right